packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java

```diff
@@ -213,25 +225,66 @@ static Integer convertMediaType(String mediaType) throws IllegalArgumentExceptio
     return MEDIA_TYPES.get(mediaType);
   }
 
-  static int[] getSizeFromCursor(Cursor cursor, int mediaType, int localUriIndex){
+  static int[] getSizeFromCursor(ContentResolver contentResolver, ExifInterface exifInterface, Cursor cursor, int mediaType, int localUriIndex) {
+    final String uri = cursor.getString(localUriIndex);
+
+    if (mediaType == Files.FileColumns.MEDIA_TYPE_VIDEO) {
+      Uri photoUri = Uri.parse("file://" + uri);
+      try {
+        AssetFileDescriptor photoDescriptor = contentResolver.openAssetFileDescriptor(photoUri, "r");
+        MediaMetadataRetriever retriever = new MediaMetadataRetriever();
+        try {
+          retriever.setDataSource(photoDescriptor.getFileDescriptor());
+          int videoWidth = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH)
+          );
+          int videoHeight = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT)
+          );
+          int videoOrientation = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION)
+          );
+          return maybeRotateAssetSize(videoWidth, videoHeight, videoOrientation);
+        } catch (NumberFormatException e) {
+        } finally {
+          retriever.release();
+          photoDescriptor.close();
+        }
+      } catch (Exception e) {
+      }
+    }
+
+    final int widthIndex = cursor.getColumnIndex(MediaStore.MediaColumns.WIDTH);
+    final int heightIndex = cursor.getColumnIndex(MediaStore.MediaColumns.HEIGHT);
     final int orientationIndex = cursor.getColumnIndex(Media.ORIENTATION);
-    final int widthIndex = cursor.getColumnIndex(Media.WIDTH);
-    final int heightIndex = cursor.getColumnIndex(Media.HEIGHT);
-
-    int[] size;
-    // If image doesn't have the required information, we can get them from Bitmap.Options
-    if ((cursor.getType(widthIndex) == Cursor.FIELD_TYPE_NULL ||
-        cursor.getType(heightIndex) == Cursor.FIELD_TYPE_NULL) &&
-        mediaType == Files.FileColumns.MEDIA_TYPE_IMAGE) {
```

Comment:
Is it safe to omit these conditions?

Reply:
Oops, good catch. The `getType` checks shouldn't be necessary, but we should keep the `mediaType` check, if only to avoid the perf hit of a useless check